src/elysium_prometheus.erl

```diff
@@ -0,0 +1,44 @@
+-module(elysium_prometheus).
+-author("leo").
+
+%% API
+-export([setup/0]).
+-export([report_metrics/3]).
+
+
+-define(DEFAULT_REGISTRY, default).
+
+setup() ->
+    is_prometheus_started() andalso do_setup().
+
+is_prometheus_started() ->
+    [AppName || {AppName, _, _} <- application:which_applications(), AppName == prometheus] == [prometheus].
```

Comment:
application:which_application() will add some latency  , we should look into minimizing it.

(xmpp@env5-xmpp-16-04-i-0136f4e252dd42303.env5-internal.tigertext.xyz)5> lists:sum([element(1, timer:tc(application, which_applications, [])) || _ <- lists:seq(1,100)])/100. 
119.77

Reply:
What I here to do is to know if the "prometheus" application started or not, this information cannot be fetched by calling application:get_env() (config:get_env eventually calling application:get_env()). Besides I spawn a new process to report the metrics, so should have on impact on other operations

Reply:
you can just check a flag, if metrics collection is enabled, you can report otherwise not. by default it will be disabled ..this way you can also control the reporting per box

Reply:
Let's keep it as it is as we discussed